plugins/modules/iso_extract.py

```diff
@@ -157,7 +164,8 @@ def main():
         cmd = [binary, 'x', image, '-o%s' % tmp_dir] + extract_files
     else:
         cmd = [module.get_bin_path('mount'), '-o', 'loop,ro', image, tmp_dir]
-
+    if password:
+        cmd += "-p%s" % password
```

Comment:
Passing passwords on the command line is dangerous. Is there a way to avoid this, like passing it in via a pipe?

Reply:
the `-p` option is more secure than you would expect because when 7z runs, if it detects the `-p` argument, it will copy the value in memory and overwrite the argument value (if you check with `ps -aux`  you might see `/usr/bin/7z -p*****`

Reply:
That's the bare minimum that a program should do if it accepts passwords on the command line. There's still a very short amount of time between invocation and 7z cleaning the command line when the password is visible.

Reply:
If there's no better way, you should add warnings to the documentation that there is a chance that the password is revealed.